Command/IndexCommand.php

```diff
@@ -59,6 +59,8 @@ public function configure()
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         $this->queue->listen(function ($message) {
+            $data = json_decode($message->body, true);
+            $this->indexer->setMetadata($data['metadata']);
             $this->indexer->prepareDocument($message);
```

Comment:
I think the metadata should be a seperate argument for the prepareDocument function instead of a standalone function. Since the metadata you receive from RabbitMQ is only counting for one document.